client/src/components/EditProfileForm.js

```diff
@@ -0,0 +1,278 @@
+    {
+      component: TextField,
+      name: "name",
+      label: "Full Name",
+      inputClass: "largeWidth",
+      defaultValue: user.name,
+      validation: {
+        required: "Name is required.",
+        maxLength: { value: 50, message: "Too many characters (max: 50)." },
+      },
+    },
+    {
+      component: TextField,
+      name: "aboutMe",
+      label: "About Me",
+      inputClass: "largeWidth",
+      defaultValue: user.aboutMe,
+      multiline: true,
+      rows: 5,
+      validation: {
+        required: "Please provide a description about you",
+      },
+    },
+    {
+      component: TextField,
+      name: "address",
+      label: "Address",
+      defaultValue: user.address,
+      inputClass: "largeWidth",
+      validation: {
+        required: "Address is required.",
+      },
+    },
+    {
+      component: TextField,
+      name: "city",
+      label: "City",
+      defaultValue: user.city,
+      inputClass: "mediumWidth",
+      validation: {
+        required: "City is required.",
+      },
+    },
+    {
+      component: TextField,
+      name: "state",
+      label: "State",
+      defaultValue: user.state,
+      inputClass: "smallWidth",
+    },
+    {
+      component: TextField,
+      name: "zipcode",
+      label: "Zipcode",
+      defaultValue: user.zipcode,
+      inputClass: "smallWidth",
+      validation: {
+        required: "Zipcode is required.",
+      },
+    },
+    {
+      component: TextField,
+      inputClass: "mediumWidth",
+      name: "country",
+      label: "Country",
+      defaultValue: user.country,
+      validation: {
+        required: "Country is required.",
+      },
+    },
+  ];
+  const cuisines = [
+    "Japanese",
+    "Martian",
+    "American",
+    "Chinese",
+    "Middle Eastern",
+  ];
+
+  return (
+    <>
+      <Typography component="h1" variant="h5">
+        Edit your profile
+      </Typography>
+      <form
+        onSubmit={handleSubmit(onSubmit)}
+        className={classes.form}
+        noValidate
+      >
+        {fields.map(
+          (
+            {
+              component: Component = TextField,
+              inputClass = "",
+              name,
+              label,
+              type = "text",
+              defaultValue,
+              validation,
+              multiline = false,
+              rows,
+            },
+            index
+          ) => (
+            <FormControl key={index}>
+              <Component
+                variant="outlined"
+                margin="normal"
+                required
+                id={name}
+                className={classes[inputClass]}
+                label={label}
+                name={name}
+                autoComplete={name}
+                type={type}
+                multiline={multiline}
+                rows={rows}
+                defaultValue={defaultValue}
+                autoFocus={index === 0}
+                inputRef={register(validation)}
+              />
```

Comment:
I think we wanted to have fullWidth as well on for these components (so that they expand the whole dialog)

We also wanted to use Grids I believe for some of the components so that they can expand equally the width of the dialog